packages/esm-patient-registration-app/src/patient-registration/field/address/address-fields.component.tsx

```diff
@@ -83,11 +92,84 @@ export const AddressHierarchy: React.FC = () => {
     setAddressLayout(propertiesObj);
   }, [t, addressTemplateXml, setFieldValue, values]);
 
+  const orderedAddressFields = useMemo(() => {
+    if (isLoadingFieldOrder || errorFetchingFieldOrder) {
+      return [];
+    }
+
+    const orderMap = Object.fromEntries(orderedFields.map((field, indx) => [field, indx]));
+
+    return [...addressLayout].sort(
+      (existingField1, existingField2) => orderMap[existingField1.name] - orderMap[existingField2.name],
+    );
+  }, [isLoadingFieldOrder, errorFetchingFieldOrder, orderedFields, addressLayout]);
+
   if (!addressTemplate) {
     return (
       <div>
         <h4 className={styles.productiveHeading02Light}>{t('addressHeader', 'Address')}</h4>
-        <SkeletonText />
+        <div
+          style={{
```

Comment:
Frankly, since we have it repeated four times, it seems worth having a component (defined in this file) that just renders:

```jsx
const AddressInputSection = ({children}) => (
<div>
  <h4 className={styles.productiveHeading02Light}>{t('addressHeader', 'Address')}</h4>
  {children}
</div>
);
```

And then use it like:

```jsx
<AddressInputSection>
  <SkeletonText />
</AddressInputSection>

<AddressInputSection>
    {addressLayout.map((attributes, index) => (
      <Input
        key={`combo_input_${index}`}
        name={`address.${attributes.name}`}
        labelText={t(attributes.label)}
        id={attributes.name}
        setSelectedValue={setSelectedValue}
        selected={selected}
      />
    ))}
</AddressInputSection>

// etc.
```

This would be somewhat more efficient, because this structure will enable React to not need to re-create the wrapping div and h4 elements in response to state changes.

Reply:
Thanks @ibacher !
I'll go with this!
Thanks!